### PR TITLE
Centcom expansion

### PR DESCRIPTION
**Pull request by noudoit**
## About the PR 
this PR adds more uniforms and items to Centcom such as Centcom gloves hats and capes/mantles they will be added to a job spawn that will have restrictions once @Chief-Engineer implements it.
**Media**

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase
Screenshot 2023-01-11 140937
Screenshot 2023-01-11 141755
Screenshot 2023-01-11 143658
Screenshot 2023-01-11 143553
Screenshot 2023-01-11 143531
Screenshot 2023-01-11 143505
Screenshot 2023-01-11 143454
Screenshot 2023-01-11 143444

**Changelog**

:cl:
- add: added Centcom drip

### CODE REVIEW

**Comment by noudoit** [comment]
still updating just wanted to PR it really quick

**Comment by noudoit** [comment]
fixing mantles (:

**Comment by noudoit** [comment]
Why merge conflicts why

File: Resources/Locale/en-US/job/job-description.ftl [hunk 1]
```diff
@@ -8,6 +8,8 @@ job-description-captain = Keep the station running, delegate work to the other h
 job-description-cargotech = Deal with requisitions and deliveries for emergencies, pilot the cargo shuttle, and work with others to make ludicrous amounts of cash.
 job-description-ce = Manage the engineering department to ensure power, atmospherics, and the hull are in perfect shape.
 job-description-centcomoff = Act as an ambassador to the newest state-of-the-art space station in Nanotrasen's fleet.
+job-description-centcom-admiral = Waste resources with rediculous demands as inters work tirelessly to attempt to justify those demands.
```
Comment (AJCM-git) [inline Resources/Locale/en-US/job/job-description.ftl:11]: ridiculous*  interns*

File: Resources/Prototypes/Entities/Clothing/Neck/mantles.yml [hunk 2]
```diff
@@ -62,4 +62,47 @@
...
+
+- type: entity
+  parent: ClothingNeckBase
+  id: ClothingNeckMantleMajor
+  suffix: DO NOT MAP
+  name: Major's mantle
+  description: A red padded pauldron signifying the rank of Major.
+  components:
+  - type: Tag
+    tags: [] # ignore "WhitelistChameleon" tag
```
Comment (AJCM-git) [inline Resources/Prototypes/Entities/Clothing/Neck/mantles.yml:103]: im pretty sure something like this already overwrites `WhitelistChamaleon` (do this to everything else)
```suggestion
  - type: Tag # ignore "WhitelistChameleon" tag
```

File: Content.Server/Clothing/Systems/RestrictedClothingSystem.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+        bool applyEffect = !mind.HasMind || mind.Mind!.Session == null || component.RequireWhitelist;
+
+        if (!applyEffect)
+            foreach (var perm in component.Permissions)
+            {
+                if (!Enum.TryParse<AdminFlags>(perm, out var flag))
+                    continue;
+
+                if (_adminManager.HasAdminFlag(mind.Mind!.Session!, flag, true))
```
Comment (AJCM-git) [inline Content.Server/Clothing/Systems/RestrictedClothingSystem.cs:51]: Handle `Mind` and `Session` nullability

File: Content.Server/Clothing/Components/RestrictedClothingComponent.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
+﻿using Content.Server.Speech.Components;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+
+namespace Content.Server.Clothing.Components;
+
+/// <summary>
+///     Does bad things when someone tries to wear something they shouldn't
+/// </summary>
+[RegisterComponent]
+public sealed class RestrictedClothingComponent : Component
```
Comment (AJCM-git) [inline Content.Server/Clothing/Components/RestrictedClothingComponent.cs:10]: Access attribute

File: Content.Server/Clothing/Systems/RestrictedClothingSystem.cs [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+    private void OnGotEquipped(EntityUid uid, RestrictedClothingComponent component, GotEquippedEvent args)
+    {
+        if (!TryComp(uid, out ClothingComponent? clothing))
+            return;
+
+        // check if entity was actually used as clothing
+        // not just taken in pockets or something
+        var isCorrectSlot = clothing.Slots.HasFlag(args.SlotFlags);
+        if (!isCorrectSlot) return;
```
Comment (AJCM-git) [inline Content.Server/Clothing/Systems/RestrictedClothingSystem.cs:35]:
```suggestion
        if (!clothing.Slots.HasFlag(args.SlotFlags) 
        	return;
```

File: Content.Server/Clothing/Components/RestrictedClothingComponent.cs [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    ///     List of permissions required to be allowed to wear the item
+    /// </summary>
+    [DataField("permissions")] [ViewVariables(VVAccess.ReadWrite)]
+    public HashSet<string> Permissions = new();
+
+    /// <summary>
+    ///     Uid that won't be affected
+    /// </summary>
+    [DataField("whitelistedUid")] [ViewVariables(VVAccess.ReadWrite)]
+    public int WhitelistedUid = 0;
```
Comment (AJCM-git) [inline Content.Server/Clothing/Components/RestrictedClothingComponent.cs:22]: shouldnt this be an `EntityUid` ?

File: Content.Server/Clothing/Components/RestrictedClothingComponent.cs [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+///     Does bad things when someone tries to wear something they shouldn't
+/// </summary>
+[RegisterComponent]
+public sealed class RestrictedClothingComponent : Component
+{
+    /// <summary>
+    ///     List of permissions required to be allowed to wear the item
+    /// </summary>
+    [DataField("permissions")] [ViewVariables(VVAccess.ReadWrite)]
+    public HashSet<string> Permissions = new();
```
Comment (AJCM-git) [inline Content.Server/Clothing/Components/RestrictedClothingComponent.cs:16]: cant you make this a hashset of admin flags and then you dont have to parse it in the system?

**Review by AJCM-git — CHANGES_REQUESTED** [review]
Resolve conflicts, replace the "Do not map" with "Admin".

File: Content.Server/Clothing/Components/RestrictedClothingComponent.cs (hunk 4 above)
Comment (metalgearsloth) [reply Content.Server/Clothing/Components/RestrictedClothingComponent.cs:10]: Also predict it

File: Content.Server/Clothing/Components/RestrictedClothingComponent.cs (hunk 7 above)
Comment (metalgearsloth) [reply Content.Server/Clothing/Components/RestrictedClothingComponent.cs:16]: AdminFlags is a flag so you can just use public AdminFlags and mask it.

File: Content.Server/Clothing/Components/RestrictedClothingComponent.cs (hunk 6 above)
Comment (metalgearsloth) [reply Content.Server/Clothing/Components/RestrictedClothingComponent.cs:22]: Hacky code should be documented as such but in this case it should still be updated.

File: Content.Server/Clothing/Systems/RestrictedClothingSystem.cs [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+        // check if entity was actually used as clothing
+        // not just taken in pockets or something
+        if (!clothing.Slots.HasFlag(args.SlotFlags) 
+        	return;
+
+        if (!TryComp(args.Equipee, out MindComponent? mind))
+            return;
+
+        if (args.Equipee.GetHashCode() == component.WhitelistedUid)
+            return;
```
Comment (metalgearsloth) [inline Content.Server/Clothing/Systems/RestrictedClothingSystem.cs:41]: See above.

File: Content.Server/Clothing/Systems/RestrictedClothingSystem.cs [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+        bool applyEffect = !mind.HasMind || mind.Mind!.Session == null || component.RequireWhitelist;
+
+        if (!applyEffect)
+            foreach (var perm in component.Permissions)
+            {
+                if (!Enum.TryParse<AdminFlags>(perm, out var flag))
+                    continue;
```
Comment (metalgearsloth) [inline Content.Server/Clothing/Systems/RestrictedClothingSystem.cs:49]: This indentation is cooked.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Server/Clothing/Systems/RestrictedClothingSystem.cs (hunk 3 above)
Comment (AJCM-git) [reply Content.Server/Clothing/Systems/RestrictedClothingSystem.cs:51]: Then you can remove the null forgiving operators, if the IDE complains there must be something wrong (`Mind` has a `TryGetSession()` method btw)

File: Content.Server/Clothing/Systems/RestrictedClothingSystem.cs [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+using Content.Shared.Administration;
+using Content.Shared.Clothing.Components;
+using Content.Shared.Inventory.Events;
+using Content.Shared.Popups;
+using Robust.Shared.Player;
+
+namespace Content.Server.Clothing.Systems;
+
+public sealed class RestrictedClothingSystem : EntitySystem
+{
```
[the remark this answers is not in the document]
Comment (metalgearsloth) [reply Content.Server/Clothing/Systems/RestrictedClothingSystem.cs:16]: Unless it actively mispredicts better to cause less work in future.

**Comment by metalgearsloth** [comment]
@noudoit you coming back to this.

File: Content.Server/Clothing/Components/RestrictedClothingComponent.cs (hunk 6 above)
Comment (metalgearsloth) [reply Content.Server/Clothing/Components/RestrictedClothingComponent.cs:22]: Permissions duplicates whitelist so requireswhitelist still needs purging.

**Comment by AJCM-git** [comment]
are you coming back to this?

**Pull request closed without merging**